packages/palette/src/elements/RadioGroup/RadioGroup.tsx

```diff
@@ -22,94 +26,119 @@ export interface RadioGroupProps extends FlexProps {
+    inputModality.current = "keyboard"
+
+    switch (event.key) {
+      case "ArrowDown":
+      case "ArrowRight":
+        handleNext()
+        break
+      case "ArrowUp":
+      case "ArrowLeft":
+        handlePrev()
+        break
     }
   }
 
-  onSelect = ({ value }) => {
-    // After state update, call back up the tree with the latest state
-    const update = () => {
-      if (this.props.onSelect) {
-        this.props.onSelect(this.state.selectedOption)
-      }
+  useEffect(() => {
+    setSelectedOption(defaultValue)
+  }, [defaultValue])
+
+  useEffect(() => {
+    if (onSelect) {
+      onSelect(selectedOption)
     }
+  }, [selectedOption])
 
-    if (this.props.deselectable) {
-      if (this.state.selectedOption === value) {
-        this.setState(
-          {
-            selectedOption: null,
-          },
-          update
-        )
-        return
-      }
+  useEffect(() => {
+    if (inputModality.current === "mouse") return
+
+    const timeout = setTimeout(() => {
+      nodes[focusIndex].ref.current?.focus()
+    }, 0)
+
+    return () => {
+      clearTimeout(timeout)
     }
+  }, [nodes, focusIndex])
 
-    this.setState({ selectedOption: value }, update)
-  }
+  return (
+    <Flex flexDirection="column" role="radiogroup" {...rest}>
+      {disabled && disabledText && (
+        <Text variant="small" mb={0.3} color="black60">
+          {disabledText}
+        </Text>
+      )}
 
-  renderRadioButtons() {
-    return React.Children.map(
-      this.props.children,
-      (child: React.ReactElement<RadioProps>) => {
+      {nodes.map(({ child, ref }) => {
         return React.cloneElement(child, {
+          ref,
+          selected: selectedOption === child.props.value,
+          tabIndex: selectedOption === child.props.value ? 0 : -1,
+          onKeyDown: handleKeydown,
           disabled:
             child.props.disabled !== undefined
               ? child.props.disabled
-              : this.props.disabled,
+              : disabled,
           onSelect: child.props.onSelect
-            ? selected => {
-                this.onSelect(selected)
+            ? (selected: { selected: boolean; value: string }) => {
+                handleSelect(selected)
                 child.props.onSelect(selected)
               }
-            : this.onSelect,
-          // FIXME: Throw an error `child.props.selected' is set once we enable the dev code elimination.
-          selected: this.state.selectedOption === child.props.value,
+            : handleSelect,
         })
-      }
-    )
-  }
-
-  render() {
-    const {
-      disabled,
-      disabledText,
-      onSelect,
-      defaultValue,
-      children,
-      ...others
-    } = this.props
-    return (
-      <Flex flexDirection="column" {...others}>
-        {disabled && disabledText && (
-          <Sans size="2" my={0.3} color="black60">
-            {disabledText}
-          </Sans>
-        )}
-        {this.renderRadioButtons()}
-      </Flex>
```

Comment:
This disabled label is kinda weird — like why wouldn't you just have a text component outside of this; not sure why it's in the RadioGroup component.